### PR TITLE
Script checks both locations for files 

### PR DESCRIPTION
**Pull request by ghost**
and does not attempt to work if nothing needs doing.

### CODE REVIEW

File: rsync-downloaded-files.sh [hunk 1]
```diff
@@ -1,21 +1,41 @@
 #!/bin/bash
 
+set -e
+
 # Iterate over all the collections that are fully loaded into the database
 
 for collection in $(psql -U ocdskfpreadonly -d ocdskingfisherprocess -h localhost -c "select source_id, to_char(data_version, 'YYYYMMDD_HH24MISS') from collection where store_end_at is not null and transform_type is null" -qAtX | tr "|" "/"); do
 
+        echo "COLLECTION ${collection}"
```
Comment (robredpath) [inline rsync-downloaded-files.sh:9]: This script was originally attended to be run via cron. Probably best to keep it silent if there's no tty, unless this was intentional

File: rsync-downloaded-files.sh [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
-        # Rsync to the disk server
-	ssh -o StrictHostKeyChecking=no archive@archive.kingfisher.open-contracting.org "mkdir -p /home/archive/data/${collection}"
 
-        rsync -r -e "ssh -o StrictHostKeyChecking=no" /home/ocdskfs/scrapyd/data/${collection}/ archive@archive.kingfisher.open-contracting.org:/home/archive/data/${collection}/ || continue
+        # Find folder on source server
+	# LOCAL_DIR serves 2 purposes; the data may be in one of 2 places AND more importantly, if the data has already been successfully archived, nothing is done.
+	LOCAL_DIR=""
+	if [ -d "/home/ocdskfs/scrapyd/data/${collection}/" ]; then 
+		LOCAL_DIR="/home/ocdskfs/scrapyd/data/${collection}/"
+	elif [ -d "/home/ocdskfs/old-data/${collection}" ]; then
```
Comment (robredpath) [inline rsync-downloaded-files.sh:20]: Will we be using `old-data` over the long term? Or is there a one-time cleanup job we can do?

File: rsync-downloaded-files.sh [hunk 3]
```diff
@@ -1,21 +1,41 @@
...
+	if [ -d "/home/ocdskfs/scrapyd/data/${collection}/" ]; then 
+		LOCAL_DIR="/home/ocdskfs/scrapyd/data/${collection}/"
+	elif [ -d "/home/ocdskfs/old-data/${collection}" ]; then 
+		LOCAL_DIR="/home/ocdskfs/old-data/${collection}/"
+	fi
+	
+        if [ "${LOCAL_DIR}" ]; then
+
+		# Create folder on disk server
+		ssh -o StrictHostKeyChecking=no archive@archive.kingfisher.open-contracting.org "mkdir -p /home/archive/data/${collection}"
```
Comment (robredpath) [inline rsync-downloaded-files.sh:27]: Not a blocker now, but the `StrictHostKeyChecking=no` was just to get around chicken/egg with Salt development. Keys and auth_keys really should be managed by salt.

File: rsync-downloaded-files.sh [hunk 4]
```diff
@@ -1,21 +1,41 @@
...
 
-        # Delete original files if rsync returned 0 (restricted to the right place by sudoers)
-        sudo -u ocdskfs rm -rf /home/ocdskfs/scrapyd/data/${collection}
+	        # Rsync to the disk server
+		rsync -r -e "ssh -o StrictHostKeyChecking=no" ${LOCAL_DIR} archive@archive.kingfisher.open-contracting.org:/home/archive/data/${collection}/ || continue
 
-	# Compress target files
-        ssh archive@archive.kingfisher.open-contracting.org "tar -Jcvf /home/archive/data/${scraper}_${timestamp}.tar.xz /home/archive/data/$collection && rm -rf /home/archive/data/${collection}"
+		# Delete original files if rsync returned 0 (restricted to the right place by sudoers)
```
Comment (robredpath) [inline rsync-downloaded-files.sh:32]: Have you checked how specific `sudoers` is?

**Review by robredpath — APPROVED** [review]
Looks good, lots of minor comments but happy for it to be merged if they're not relevant

File: rsync-downloaded-files.sh (hunk 1 above)
Comment (ghost) [reply rsync-downloaded-files.sh:9]: There is already some output, so I thought we might as well go for some more. We can look at making sure it's logged somewhere nice - that's probably the right thing to do for debugging purposes anyway!

File: rsync-downloaded-files.sh (hunk 2 above)
Comment (ghost) [reply rsync-downloaded-files.sh:20]: old-data is the scrapers that have yet to be ported to Scrapy and run in a cut down version of the old kingfisher. As there is no scheduled work on porting them at the moment, there is no set date for them finishing but hopefully not long term. But as far as I know, this is the only issue this creates and an if statement solved it.

File: rsync-downloaded-files.sh (hunk 3 above)
Comment (ghost) [reply rsync-downloaded-files.sh:27]: Noted, thanks

File: rsync-downloaded-files.sh (hunk 4 above)
Comment (ghost) [reply rsync-downloaded-files.sh:32]: Yes, and I've got a commit ready to add the second location.

**Comment by ghost** [comment]
They are relevant and are noted, but in the ethos of "small steps forward" I'll merge this and start testing and looking at some of the other issues! Thanks

**Pull request merged**